test/hooks/WETHHook.t.sol

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {Test} from "forge-std/Test.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolSwapTest} from "@uniswap/v4-core/src/test/PoolSwapTest.sol";
+import {CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {TickMath} from "@uniswap/v4-core/src/libraries/TickMath.sol";
+import {Deployers} from "@uniswap/v4-core/test/utils/Deployers.sol";
+import {CustomRevert} from "@uniswap/v4-core/src/libraries/CustomRevert.sol";
+import {WETH} from "solmate/src/tokens/WETH.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Hooks} from "@uniswap/v4-core/src/libraries/Hooks.sol";
+import {Test} from "forge-std/Test.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {MockERC20} from "solmate/src/test/utils/mocks/MockERC20.sol";
+
+import {BaseTokenWrapperHook} from "../../src/base/hooks/BaseTokenWrapperHook.sol";
+import {WETHHook} from "../../src/hooks/WETHHook.sol";
+
+contract WETHHookTest is Test, Deployers {
+    using PoolIdLibrary for PoolKey;
+    using CurrencyLibrary for Currency;
+
+    WETHHook public hook;
+    WETH public weth;
+    PoolKey poolKey;
+    uint160 initSqrtPriceX96;
+
+    // Users
+    address payable alice = payable(makeAddr("alice"));
+    address payable bob = payable(makeAddr("bob"));
+
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    function setUp() public {
+        deployFreshManagerAndRouters();
+        weth = new WETH();
+
+        // Deploy WETH hook
+        hook = WETHHook(
+            payable(
+                address(
+                    uint160(
+                        type(uint160).max & clearAllHookPermissionsMask | Hooks.BEFORE_SWAP_FLAG
+                            | Hooks.BEFORE_ADD_LIQUIDITY_FLAG | Hooks.BEFORE_SWAP_RETURNS_DELTA_FLAG
+                            | Hooks.BEFORE_INITIALIZE_FLAG
+                    )
+                )
+            )
+        );
+        deployCodeTo("./foundry-out/WETHHook.sol/WETHHook.default.json", abi.encode(manager, weth), address(hook));
```

Comment:
ooh whats the difference between this and etch? 👀  ive always used etch, never seen this function before

Reply:
etch you have to have the bytecode inline. best way to get that was to deploy a dummy version and then etch the bytecode over to the right address. but that the dummy version reverts on hook validation - so this lets me skip the dummy deployments tep

Reply:
oh smarrrrt